sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDBStateEncoder.scala

```diff
@@ -24,15 +24,18 @@ import java.nio.{ByteBuffer, ByteOrder}
 import org.apache.spark.internal.Logging
 import org.apache.spark.sql.catalyst.expressions.{BoundReference, JoinedRow, UnsafeProjection, UnsafeRow}
 import org.apache.spark.sql.catalyst.expressions.codegen.UnsafeRowWriter
-import org.apache.spark.sql.execution.streaming.state.RocksDBStateStoreProvider.{STATE_ENCODING_NUM_VERSION_BYTES, STATE_ENCODING_VERSION}
+import org.apache.spark.sql.execution.streaming.state.RocksDBStateStoreProvider.{STATE_ENCODING_NUM_VERSION_BYTES, STATE_ENCODING_VERSION, VIRTUAL_COL_FAMILY_PREFIX_BYTES}
 import org.apache.spark.sql.types._
 import org.apache.spark.unsafe.Platform
 
 sealed trait RocksDBKeyStateEncoder {
   def supportPrefixKeyScan: Boolean
   def encodePrefixKey(prefixKey: UnsafeRow): Array[Byte]
+  // TODO try change less of the API for key encoder?
+  def encodePrefixKey(prefixKey: UnsafeRow, colFamilyId: Short): Array[Byte]
   def encodeKey(row: UnsafeRow): Array[Byte]
-  def decodeKey(keyBytes: Array[Byte]): UnsafeRow
+  def encodeKey(row: UnsafeRow, colFamilyId: Short): Array[Byte]
```

Comment:
maybe we can just combine this into

`def encodeKey(row: UnsafeRow, vcfId: Option[Short] = None)` ?